pyproject.toml

```diff
@@ -58,28 +58,27 @@ progressbar2 = "^3.55.0"
 PyYAML = ">=5.1,<7.0"
 scikit-learn = "^1.0.1"
 scipy = "^1.7.2"
-SQLAlchemy = "<1.4.0,>=1.3.0"
+SQLAlchemy = ">=1.3,<2.0"
 numpy = ">=1.22.0,<2.0"
 jinjasql = "^0.1.8"
-ipython = "^7.29.0"
+ipython = ">=8.0.1"
 matplotlib = "^3.5.0"
 pyarrow = { version = "6.0.0", optional = true }
 cx-Oracle = { version = "^8.3.0", optional = true }
 PyHive = { version = ">=0.6.1", optional = true }
-thrift = { version = "^0.15.0", optional = true }
+thrift = { version = "0.16.0", optional = true }
 psycopg2-binary = { version = "^2.9.2", optional = true }
 PyMySQL = { version = "^1.0.2", optional = true }
 pymssql = { version = "^2.2.2", optional = true }
 pymongo = { version = "^3.12.1", optional = true }
 snowflake-connector-python = { version = "^2.7.1", optional = true }
 cryptography = { version = "3.4.7", optional = true }
-snowflake-sqlalchemy = { version = "1.2.4", optional = true }
+snowflake-sqlalchemy = { version = ">=1.2.4, <=1.4.4, !=1.2.5", optional = true } # https://github.com/snowflakedb/snowflake-sqlalchemy/issues/234
```

Comment:
Avoiding 1.2.5 because of incompatibility with sqlalchemy 1.3 ([source](https://github.com/snowflakedb/snowflake-sqlalchemy/issues/234)).